tests/phpunit/tests/functions/wpGetArchives.php

```diff
@@ -204,4 +204,41 @@ public function test_wp_get_archives_post_type() {
 		);
 		$this->assertSame( $expected, trim( $archives ) );
 	}
+
+	/**
+	 * @ticket 64304
+	 */
+	public function test_wp_get_archives_args_filter() {
+		// Test that the filter can modify the limit argument.
+		$filter_callback = function ( $args ) {
+			$args['limit'] = 3;
+			return $args;
+		};
+		add_filter( 'wp_get_archives_args', $filter_callback );
+
+		$ids = array_slice( array_reverse( self::$post_ids ), 0, 3 );
+
+		$link1 = get_permalink( $ids[0] );
+		$link2 = get_permalink( $ids[1] );
+		$link3 = get_permalink( $ids[2] );
+
+		$title1 = get_post( $ids[0] )->post_title;
+		$title2 = get_post( $ids[1] )->post_title;
+		$title3 = get_post( $ids[2] )->post_title;
+
+		$expected = <<<EOF
+<li><a href='$link1'>$title1</a></li>
+	<li><a href='$link2'>$title2</a></li>
+	<li><a href='$link3'>$title3</a></li>
+EOF;
+		$archives = wp_get_archives(
+			array(
+				'echo'  => false,
+				'type'  => 'postbypost',
+				'limit' => 5, // This should be overridden by the filter to 3.
+			)
+		);
+
+		$this->assertSameIgnoreEOL( $expected, trim( $archives ) );
```

Comment:
This would be a good opportunity to leverage [the new `assertEqualHTML()`](https://make.wordpress.org/core/2025/11/21/updates-to-the-html-api-in-6-9/#:~:text=No%20more%20hard%2Dcoding%20HTML%20string%20assertions%20in%20unit%20tests.): 

```suggestion
		$this->assertEqualHTML( $expected, $archives );
```

This would allow modifying the `EOF` heredoc above to have nicer formatting.